lib/product.rb

```diff
@@ -0,0 +1,23 @@
+class Product
+  attr_reader :category, :name, :unit_price, :quantity, :is_hoarded
+
+  def initialize(category, name, unit_price, quantity, is_hoarded = false)
+    @category = category
+    @name = name
+    @unit_price = unit_price
+    @quantity = quantity.to_i
+    @is_hoarded = is_hoarded
+  end
+
+  def total_price
+    @quantity.to_f * @unit_price
+  end
+
+  def is_hoarded?
+    is_hoarded
```

Comment:
call this using the instance variable since you have access to that.. without an instance variable, you're calling the method, which will call the instance variable -- it will still give you teh same thing, but you skip a step.